src/main/java/seedu/nutrigoals/logic/parser/ParserUtil.java

```diff
@@ -95,6 +98,31 @@ public static Calorie parseCalorie(String calorie) throws ParseException {
         return new Calorie(trimmedCalorie);
     }
 
+    /**
+     * Parses {@code String locationName} into a {@code Location}
+     * @param locationName name of location
+     * @return Location
+     * @throws ParseException if the given {@code locationName} is invalid.
+     */
+    public static Location parseLocation(String locationName) throws ParseException {
+        requireNonNull(locationName);
+        String trimmedLocationName = locationName.trim().toUpperCase();
+        Map<String, Location> out = new HashMap<>();
+        // ANY UPDATES HERE SHOULD UPDATE LOCATE'S MESSAGE_USAGE SORRY FOR COUPLING
+        // https://goo.gl/maps/8a2h8ciNGcc2DUxs8
+        out.put("COM2", new Location("COM2", "1.2942815638814327, 103.77410024788284"));
+        // https://goo.gl/maps/MgTnNAnMCVqokPqx9
+        out.put("S17", new Location("S17", "1.2976996370988612, 103.78060787462833"));
+        // https://goo.gl/maps/2Zxpi6xsmfUNzU2h7
+        out.put("CLB", new Location("CLB", "1.296642317024345, 103.77322870790687"));
+        // https://goo.gl/maps/TBxRtewZLVCgkEdF6
+        out.put("CLB", new Location("UHC", "1.2991034341416665, 103.77639982958577"));
```

Comment:
I think it would be better if we include more options, e.g. gym locations in halls? If the exact longitudes and latitudes are not accessible, maybe we can use the longitudes and latitudes of the different halls to estimate the distance?